### DISCUSSION TITLE
A warm welcome here

### DISCUSSION THREAD

**Discussion by axodox** [axodox/axodox-machinelearning#3] [Announcements]
Hi All,

Just set up this page so we can hopefully help each other enriching this lib. Some ideas which worth exploring:
- Implementing ControlNet support
- Adding oversized prompt (>77 tokens) support
- Adding support to CUDA or other execution providers besides DirectML
- Adding a Linux / Mac pathway in the small number of cases, where Windows API is used
- Cleaning up the code a bit (e.g. adding some more operators to tensors)
- Adding more documentation etc.
- Adding some different samplers

Anyways, I will keep doing some development as well on my own ;) But hope we can help each other making this a really great lib for everyone!

Best Regards:
Péter

**Comment by zolrath** [comment]
Amazing work Péter!
Exceedingly excited for ControlNet support. 
I've got a fairly specific use case for the Scribble model but I know Canny is well loved as well.

Excited to see where this goes!

**Comment by axodox** [comment]
I plan to look into making an Unreal Engine Plugin as well. It might still be a little early to use the AI in real-time, since it is pretty taxing to run right now. But I could see some use:

- image generating character portraits in classic CRPG style games
- have AI filters in photo mode, which make the scene look like a live action movie or stylized - ok now this would be super cool
- have cool, unique custom patterns generated for your cars in a racing game, or for your character items in a multiplayer game etc.
- have your character paint unique paintings in The Sims

Of course much of this could be just pre-created for now which would be simpler and safer (and take 2GB less space), but I think it still would be cool to have a completely unique outcome.

**Reply by zolrath** [reply]
My use case is very similar to those! Unreal Engine with "on demand" image generation by the user base off their input + the controlnet scribble input.

There are so many applications of this to empower users in games, should be fun!
